src/components/UserSelector.tsx

```diff
@@ -1,42 +1,115 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { User } from '../types/User';
+import classNames from 'classnames';
+import { PostsList } from './PostsList';
+import { Loader } from './Loader';
+import { getUserPosts } from '../api/posts';
+import { Post } from '../types/Post';
+
+type Props = {
+  users: User[];
+};
+
+export const UserSelector: React.FC<Props> = ({ users }) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [error, setError] = useState('');
+
+  const toggleDropdown = () => setIsOpen(!isOpen);
+
+  const handleUserSelect = (user: User) => {
+    setSelectedUser(user);
+    setLoading(true);
+    setIsOpen(false);
+    setError('');
+  };
+
+  useEffect(() => {
+    if (selectedUser) {
+      setLoading(true);
+
+      getUserPosts(selectedUser.id)
+        .then(userPosts => {
+          setPosts(userPosts);
+        })
+        .catch(() => {
+          setError('Something went wrong!');
```

Comment:
Consider providing a more user-friendly error message to inform users of what went wrong when fetching posts.